spec/lib/mastodon/cli/search_spec.rb

```diff
@@ -4,5 +4,79 @@
 require 'mastodon/cli/search'
 
 describe Mastodon::CLI::Search do
+  subject { cli.invoke(action, arguments, options) }
+
+  let(:cli) { described_class.new }
+  let(:arguments) { [] }
+  let(:options) { {} }
+
   it_behaves_like 'CLI Command'
+
+  describe '#deploy' do
+    let(:action) { :deploy }
+
+    context 'with concurrency out of range' do
+      let(:options) { { concurrency: -100 } }
+
+      it 'Exits with error message' do
+        expect { subject }
+          .to output_results('this concurrency setting')
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'with batch size out of range' do
+      let(:options) { { batch_size: -100_000 } }
+
+      it 'Exits with error message' do
+        expect { subject }
+          .to output_results('this batch_size setting')
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'without options' do
+      before { stub_search_indexes }
+
+      let(:indexed_count) { 1 }
+      let(:deleted_count) { 2 }
+
+      it 'reports about storage size' do
+        expect { subject }
+          .to output_results(
+            "Indexed #{described_class::INDICES.size * indexed_count} records",
+            "de-indexed #{described_class::INDICES.size * deleted_count}"
+          )
+      end
+    end
+
+    def stub_search_indexes
+      described_class::INDICES.each do |index|
+        allow(index)
+          .to receive_messages(
+            specification: instance_double(Chewy::Index::Specification, changed?: true, lock!: nil),
+            purge: nil
+          )
```

Comment:
The issue here was the various `...Index` classes descend from `Chewy::Index` and that's where the methods are, so that's why I did it that way. If we wanted to keep the stub more narrowly on "our" index classes, one thing I can think of is to set up delegation methods on the index classes so that methods like `specification_changed?` and `specification_lock!` could be more cleanly mocked in the spec, and just delegate to the `specification` from within the class definition.

Reply:
Don't mind me, I completely missed where we used these methods. This is fine.